### PR TITLE
libressl-devel: re-fix build on macOS < 10.14

### PR DESCRIPTION
**Pull request by aeiouaeiouaeiouaeiouaeiouaeiou**
Backport of the fix added in #26827. Once again, it fails to test correctly due to a conflict with OpenSSL.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6.8 10K549 x86_64
Xcode 4.2 4C199

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Review by artkiver — APPROVED** [review]
Looks good to me! Tested OK on 15.1.1!

The Port Health for security/libressl is already looking better on https://ports.macports.org/port/libressl/details/ too!

I still see red Xs for El Capitan, Lion and Snow Leopard, but maybe the Port Health checks haven't re-run for those systems yet? I'm not too sure how Port Health checks are generated.

Thank you again for your improvements!

Also, @aeiouaeiouaeiouaeiouaeiouaeiou you had mentioned difficulty testing, in my own experience anecdotally, (and since I am a maintainer for other things mostly down streamed from OpenBSD which also uses LibreSSL extensively) I tend to find good success starting with a fresh MacPorts installation (or removing all installed MacPorts with something like sudo port -fp uninstall installed [you probably don't want to do this on any system you are using for non testing purposes; following the instructions for the upgrade/migration would probably be more useful in such instances e.g. https://trac.macports.org/wiki/Migration ]).

Then my typical order of operations is something like:

sudo port -d -v install libressl

then install other things, e.g. openssh, opensmtpd, etc.

A lot of Portfiles have something like:

    depends_lib         path:lib/libssl.dylib:openssl \


Which in theory: should utilize whatever TLS library is installed via MacPorts (e.g. if you already have LibreSSL installed, it will attempt to use that, if OpenSSL3 is already installed, then that).

Not every Portfile uses the dylib declaration and I haven't done due diligence to index and update them all (I think it's estimated maybe 800 some odd Portfiles in total have TLS library dependencies if I am remembering a Trac issue correctly?).

In the instance of rpki-client, @neverpanic had offered some "smarts" which while similar, aren't quite the same. I tried to replicate something similar with devel/got (GoT is also from OpenBSD developers and works without issue with LibreSSL, but if using OpenSSL got-portable in more recent versions added a requirement for libretls [which is essentially sort of like backporting some things to OpenSSL from LibreSSL for lack of a better description] but my efforts to make that Portfile "smarter" failed, so I've got some comments in the Portfile for how to rectify building it with LibreSSL, while leaving it in a manner where it will still build with the GitHub Actions CI checks without failing in its defaults. That could definitely be improved no doubt, but I haven't revisited it any time recently. It was my hope to get mpbb running locally so I could attempt to replicate or at least improve my own local testing, but I have had a couple of failed attempts there so far and am not sure what part of the instructions aren't working for me and haven't focused on clearing that up yet either.

Anyway, TL;DR maybe some of that may prove useful in your own testing? Regardless, thank you again for improving the reliability of LibreSSL builds on older versions of OS X!

**Comment by aeiouaeiouaeiouaeiouaeiouaeiou** [comment]
@reneeotten @herbygillot
